colorcoder/ColorCode.java

```diff
@@ -0,0 +1,24 @@
+package colorcoder;
+
+public class ColorCode {
+	static int GetColorCodeFromColor(MajorColor major, MinorColor minor) {
+		int Colorcode = 0;
+		int MinorColorIndexCount = ColorConstants.numberOfMinorColors - 1;
+		Colorcode = major.getIndex() * 1 + minor.getIndex() + 1;
+		switch (major.getIndex()) {
+		case 1:
+			Colorcode = Colorcode + MinorColorIndexCount;
+			break;
+		case 2:
+			Colorcode = Colorcode + MinorColorIndexCount * 2;
```

Comment:
You can still do away with switch case, 
case 2:
			Colorcode = Colorcode + MinorColorIndexCount * 2;
can be replaced with => Colorcode = Colorcode + MinorColorIndexCount * major.getIndex();

Please check if switch case is still required...

Reply:
Replacing the switch statement with if condition...